core/src/test/scala/tamer/utils/FakeConsumer.scala

```diff
@@ -108,18 +108,34 @@ sealed class FakeConsumer[IK, IV](
             record = new ConsumerRecord("topic", 0, offset, record.key().asInstanceOf[K], record.value().asInstanceOf[V]),
             commitHandle = (toCommit: Map[TopicPartition, Long]) =>
               (for {
-                r <- Task(new java.util.Random().nextInt(1)).map(_.longValue) // simulates the asynchronicity
-                _ <- Task(Thread.sleep(r)) // simulates the asynchronicity
```

Comment:
While it's true that this action is asynchronous, the effects often where not measurable.